environment.yml

```diff
@@ -12,18 +12,20 @@ dependencies:
   # Running HSP2
   - python =3.8*
   - scipy   # Scipy also installs numpy
-  - pandas =1.2.*  # Pandas installs most scientific Python modules, such as Numpy, etc.
-  - numba =0.53.1
-  - numpy =1.20.*  # for compatibility with numba version <0.53
-  - hdf5 =1.10.6    # latest compatible with both versions of PyTables & h5py below
-  - pytables =3.6.1
+  - pandas ==1.2.*  # Pandas installs most scientific Python modules, such as Numpy, etc.
```

Comment:
The "==" in the environment.yml means that it can't be used by conda anymore.  I wrote "process_env_yaml" in "setup.py" so that we could keep the conda formatted yaml files.  This way, as the "environment.yml" changes for conda, pip will accept those changes and just work.  That is the idea anyway.

For pip you just have to have the hdf5 libraries installed by some other means, so the hdf5 entry in the environment.yml should stay for conda, but be ignored by pip by adding to the list inside "process_env_yaml" instead.  Same with nb_conda.

The mando entry I believe needs to in the pip section since it is only available via pip.

Really want zero impact on the conda installation process.

Reply:
@timcera, I appreciate your interest in having zero impact on the conda installation process.

Good news. I just created a fresh conda environment from this file and it seems that the `==` works just fine with conda! So we can leave that as is, which simplifies `setup.py` a bit.

On the other hand, I would like to restore the `hdf5 ==1.10.6` spec for now, because my conda enviroment auto-installed the latest `hdf5 ==1.12.1`, which we haven't really tested yet. One of the main reasons for me recommending conda install was to unify and constrain the installation of hdf5 versions after @rheaphy discovered a serious performance issue when he realized that he had two different versions in two different places on his system. See https://github.com/respec/HSPsquared/issues/36#issuecomment-697107682.  So that could be an issue with the pip install approach going forward.

I'll restor `hdf5` and `nbconda` in the conda `environment.yml` and try to add them to the list of requirements to be ignored by pip.